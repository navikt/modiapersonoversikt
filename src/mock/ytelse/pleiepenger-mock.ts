import faker from 'faker/locale/nb_NO';
import moment from 'moment';

import navfaker from 'nav-faker/dist/index';
import {
    Arbeidsforhold,
    Periode,
    Pleiepengeperiode,
    Pleiepengerettighet,
    PleiepengerResponse,
    Vedtak
} from '../../models/ytelse/pleiepenger';
import { fyllRandomListe } from '../utils/mock-utils';
import { aremark } from '../person/aremark';
import { moss } from '../person/moss';
import { pleiepengerTestData } from '../../app/personside/infotabs/ytelser/pleiepenger/pleiepengerTestData';
import { backendDatoformat } from '../../utils/dateUtils';

export function getMockPleiepenger(fødselsnummer: string): PleiepengerResponse {
    if (fødselsnummer === aremark.fødselsnummer) {
        return {
            pleiepenger: [
                {
                    ...pleiepengerTestData,
                    barnet: moss.fødselsnummer
                }
            ]
        };
    }

    faker.seed(Number(fødselsnummer));
    navfaker.seed(fødselsnummer + 'pleiepenger');

    if (navfaker.random.vektetSjanse(0.3)) {
        return {
            pleiepenger: null
        };
    }

    return {
        pleiepenger: fyllRandomListe<Pleiepengerettighet>(() => getMockPleiepengerettighet(fødselsnummer), 2)
    };
}

export function getMockPleiepengerettighet(fødselsnummer: string): Pleiepengerettighet {
    faker.seed(Number(fødselsnummer));
    navfaker.seed(fødselsnummer + 'pleiepenger');

    const pleiepengeDager = 1300;
    const forbrukteDager = navfaker.random.integer(pleiepengeDager);
    const restDager = pleiepengeDager - forbrukteDager;
    return {
        barnet: navfaker.personIdentifikator.fødselsnummer(),
        omsorgsperson: fødselsnummer,
        andreOmsorgsperson: navfaker.personIdentifikator.fødselsnummer(),
        restDagerFomIMorgen: restDager,
        forbrukteDagerTomIDag: forbrukteDager,
        pleiepengedager: pleiepengeDager,
        restDagerAnvist: restDager,
        perioder: fyllRandomListe<Pleiepengeperiode>(() => getPleiepengeperiode(), 10)
    };
}

function getPleiepengeperiode(): Pleiepengeperiode {
    return {
        fom: moment(faker.date.past(2)).format(backendDatoformat),
        antallPleiepengedager: navfaker.random.integer(20),
        arbeidsforhold: fyllRandomListe<Arbeidsforhold>(() => getArbeidsforhold(), 2),
        vedtak: fyllRandomListe<Vedtak>(() => getVedtak(), 3)
    };
}

function getArbeidsforhold(): Arbeidsforhold {
    return {
        arbeidsgiverNavn: faker.company.companyName(),
        arbeidsgiverKontonr: Number(faker.finance.account(11)).toString(),
        inntektsperiode: 'Månedssats',
        inntektForPerioden: Math.round(Number(faker.finance.amount(5000, 50000))),
        refusjonTom: moment(faker.date.past(2)).format(backendDatoformat),
        refusjonstype: 'Ikke refusjon',
        arbeidsgiverOrgnr: '1234567890',
        arbeidskategori: 'Arbeidstaker'
    };
}

function getVedtak(): Vedtak {
    return {
        periode: getPeriode(),
        kompensasjonsgrad: navfaker.random.vektetSjanse(0.5) ? 100 : navfaker.random.integer(100),
        utbetalingsgrad: navfaker.random.vektetSjanse(0.5) ? 100 : navfaker.random.integer(100),
        anvistUtbetaling: moment(faker.date.past(2)).format(backendDatoformat),
        bruttobeløp: Number(faker.commerce.price()),
        dagsats: navfaker.random.integer(70),
        pleiepengegrad: navfaker.random.integer(100)
    };
}

function getPeriode(): Periode {
    const fom = moment(faker.date.past(2));
    const tom = moment(fom).add(faker.random.number(40), 'days');
    return {
        fom: fom.format(backendDatoformat),
        tom: tom.format(backendDatoformat)
    };
}
