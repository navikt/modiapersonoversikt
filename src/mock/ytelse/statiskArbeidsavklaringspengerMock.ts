export const statiskArbeidsavklaringspengerMock = {
    barnMedStonad: 2,
    barnetillegg: 1500,
    beregningsgrunnlag: 30000,
    dagsats: 500,
    kildesystem: 'KELVIN',
    periode: {
        fraOgMedDato: '2025-01-01',
        tilOgMedDato: '2025-12-31'
    },
    rettighetsType: 'AAP',
    saksnummer: '2023123456',
    status: 'LÃ˜PENDE',
    vedtakId: 'vedtak-001',
    vedtaksdato: '2023-02-01',
    opphorsAarsak: 'FERDIGBEHANDLET'
};
