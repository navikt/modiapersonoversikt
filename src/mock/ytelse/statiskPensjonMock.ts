import type { Pensjon } from 'src/models/ytelse/pensjon';

export const statiskPensjonMock: Pensjon = {
    id: '1234',
    fom: '2020-01-10',
    tom: '2020-02-21',
    enhetId: '0129',
    type: { code: 'AFP', decode: 'Afp' },
    status: { code: 'OPPRETTET', decode: 'Opprettet' }
};
