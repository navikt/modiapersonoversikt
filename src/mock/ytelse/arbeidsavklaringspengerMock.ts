import { fakerNB_NO as faker } from '@faker-js/faker';
import dayjs from 'dayjs';
import navfaker from 'nav-faker/dist/index';
import { statiskArbeidsavklaringspengerMock } from 'src/mock/ytelse/statiskArbeidsavklaringspengerMock';
import type {
    Arbeidsavklaringspenger,
    ArbeidsavklaringspengerResponse
} from 'src/models/ytelse/arbeidsavklaringspenger';
import { backendDatoformat } from 'src/utils/date-utils';
import { aremark } from '../persondata/aremark';
import { fyllRandomListe } from '../utils/mock-utils';

export function getMockArbeidsavklaringspengerResponse(fnr: string): ArbeidsavklaringspengerResponse {
    if (fnr === aremark.personIdent) {
        return [statiskArbeidsavklaringspengerMock];
    }

    faker.seed(Number(fnr));
    navfaker.seed(`${fnr}arbeidsavklaringspenger`);

    if (navfaker.random.vektetSjanse(0.3)) {
        return [];
    }

    return fyllRandomListe<Arbeidsavklaringspenger>(() => getMockArbeidsavklaringspenger(fnr), 3);
}

function getMockArbeidsavklaringspenger(fnr: string): Arbeidsavklaringspenger {
    faker.seed(Number(fnr));
    navfaker.seed(`${fnr}pensjon`);

    const fomDato = dayjs(faker.date.past({ years: 2 })).format(backendDatoformat);
    const tomDato = dayjs(fomDato).add(faker.number.int(40), 'days').format(backendDatoformat);

    return {
        barnMedStonad: 2,
        barnetillegg: 1500,
        beregningsgrunnlag: 30000,
        dagsats: 500,
        kildesystem: 'Arena',
        periode: {
            fraOgMedDato: fomDato,
            tilOgMedDato: tomDato
        },
        rettighetsType: 'BISTANDSBEHOV',
        saksnummer: '2023123456',
        status: 'LÃ˜PENDE',
        vedtakId: faker.number.int(100).toString(),
        vedtaksdato: '2023-02-01',
        opphorsAarsak: 'FERDIGBEHANDLET'
    };
}
