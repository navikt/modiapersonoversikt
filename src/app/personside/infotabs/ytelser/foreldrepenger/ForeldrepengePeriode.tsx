import * as React from 'react';
import { Foreldrepengerperiode } from '../../../../../models/ytelse/foreldrepenger';
import Utbetalinger from '../utbetalinger/Utbetalinger';
import styled from 'styled-components';
import theme from '../../../../../styles/personOversiktTheme';
import DescriptionList from '../../../../../components/DescriptionList';
import YtelserPeriode from '../felles-styling/YtelserPeriode';
import { formaterDato } from '../../../../../utils/dateUtils';
import { YtelserKeys } from '../ytelserKeys';

interface Props {
    periode: Foreldrepengerperiode;
    periodenr: number;
}

const Padding = styled.div`
    margin: ${theme.margin.px10} ${theme.margin.px20} ${theme.margin.px40};
`;

const Flex = styled.div`
    display: flex;
`;

const Stor = styled.div`
    flex-basis: 55%;
`;

const Liten = styled.div`
    flex-basis: 45%;
`;

export function convertBoolTilJaNei(verdi: boolean | null): string | null {
    switch (verdi) {
        case true:
            return 'Ja';
        case false:
            return 'Nei';
        default:
            return null;
    }
}

function ForeldrepengePeriode({ periode, periodenr }: Props) {
    const entries = {
        'Midlertidig stans': periode.midlertidigStansDato,
        Stansårsak: periode.stansårsak,
        Mødrekvote: convertBoolTilJaNei(periode.erMødrekvote),
        'Aleneomsorg Mor': convertBoolTilJaNei(periode.harAleneomsorgMor),
        'Rett til Mødrekvote': periode.rettTilMødrekvote,
        Fedrekvote: convertBoolTilJaNei(periode.erFedrekvote),
        'Aleneomsorg Far': convertBoolTilJaNei(periode.harAleneomsorgFar),
        'Rett til Fedrekvote': periode.rettTilFedrekvote
    };
    return (
        <YtelserPeriode tittel={`Periode ${periodenr} - ${formaterDato(periode.foreldrepengerFom)}`}>
            <Flex>
                <Liten>
                    <Padding>
                        <DescriptionList entries={entries} />
                    </Padding>
                </Liten>
                <Stor>
                    <Utbetalinger
                        kommendeUtbetalinger={periode.kommendeUtbetalinger}
                        historiskeUtbetalinger={periode.historiskeUtbetalinger}
                        ytelsesType={YtelserKeys.Foreldrepenger}
                    />
                </Stor>
            </Flex>
        </YtelserPeriode>
    );
}

export default ForeldrepengePeriode;
