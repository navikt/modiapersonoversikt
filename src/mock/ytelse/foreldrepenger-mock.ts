import * as faker from 'faker/locale/nb_NO';
import * as moment from 'moment';

import navfaker from 'nav-faker/dist/index';

import {
    Arbeidsforhold,
    Foreldrepengerperiode,
    ForeldrepengerResponse,
    Foreldrepengerettighet
} from '../../models/ytelse/foreldrepenger';
import { getPeriode } from '../person/periodeMock';
import { backendDatoformat, fyllRandomListe, vektetSjanse } from '../utils/mock-utils';
import { getHistoriskUtbetaling, getKommendeUtbetaling } from './ytelse-utbetalinger-mock';
import { HistoriskUtbetaling, KommendeUtbetaling } from '../../models/ytelse/ytelse-utbetalinger';

export function getMockForeldrepenger(fødselsnummer: string): ForeldrepengerResponse {
    faker.seed(Number(fødselsnummer));
    navfaker.seed(fødselsnummer + 'foreldrepenger');

    if (navfaker.random.vektetSjanse(0.3)) {
        return {
            foreldrepenger: null
        };
    }

    return {
        foreldrepenger: getForeldrepengerettighetMock(fødselsnummer)
    };
}

export function getForeldrepengerettighetMock(fødselsnummer: string): Foreldrepengerettighet {
    return {
        forelder: fødselsnummer,
        andreForeldersFnr: navfaker.personIdentifikator.fødselsnummer(),
        antallBarn: navfaker.random.integer(5),
        barnetsFødselsdato: moment(faker.date.recent()).format(backendDatoformat),
        dekningsgrad: navfaker.random.integer(95),
        fedrekvoteTom: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        mødrekvoteTom: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        foreldrepengetype: foreldrePengeType(),
        graderingsdager: navfaker.random.integer(100),
        restDager: navfaker.random.integer(50),
        rettighetFom: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        eldsteIdDato: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        foreldreAvSammeKjønn: vektetSjanse(faker, 0.5) ? 'Begge er pappaer' : null,
        periode: fyllRandomListe<Foreldrepengerperiode>(() => getForeldrepengerperiodeMock(fødselsnummer), 5),
        slutt: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        arbeidsforhold: fyllRandomListe<Arbeidsforhold>(() => getArbeidsforholdMock(fødselsnummer), 5),
        erArbeidsgiverperiode: vektetSjanse(faker, 0.5) ? faker.random.boolean() : null,
        arbeidskategori: vektetSjanse(faker, 0.5) ? 'Arbeidstaker' : null
    };
}

export function getForeldrepengerperiodeMock(fødselsnummer: string): Foreldrepengerperiode {
    return {
        fødselsnummer: fødselsnummer,
        harAleneomsorgFar: vektetSjanse(faker, 0.5) ? faker.random.boolean() : null,
        harAleneomsorgMor: vektetSjanse(faker, 0.5) ? faker.random.boolean() : null,
        arbeidsprosentMor: vektetSjanse(faker, 0.5) ? navfaker.random.integer(100) : null,
        avslagsårsak: vektetSjanse(faker, 0.5) ? 'Avslag' : null,
        avslått: vektetSjanse(faker, 0.5) ? 'Avslått' : null,
        disponibelGradering: vektetSjanse(faker, 0.5) ? navfaker.random.integer(75) : null,
        erFedrekvote: vektetSjanse(faker, 0.5) ? faker.random.boolean() : null,
        erMødrekvote: vektetSjanse(faker, 0.5) ? faker.random.boolean() : null,
        forskyvelsesårsak1: vektetSjanse(faker, 0.5) ? 'FÅRSAK1' : null,
        forskyvelsesperiode1: vektetSjanse(faker, 0.5) ? getPeriode() : null,
        forskyvelsesårsak2: vektetSjanse(faker, 0.5) ? 'FÅRSAK2' : null,
        forskyvelsesperiode2: vektetSjanse(faker, 0.5) ? getPeriode() : null,
        foreldrepengerFom: moment(faker.date.past(5)).format(backendDatoformat),
        midlertidigStansDato: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        morSituasjon: vektetSjanse(faker, 0.5) ? faker.lorem.words(5) : null,
        rettTilFedrekvote: vektetSjanse(faker, 0.5) ? 'Rett til fedrekvote' : 'Ingen rett til fedrekvote',
        rettTilMødrekvote: vektetSjanse(faker, 0.5) ? 'Rett til mødrekvote' : 'Ingen rett til mødrekvote',
        stansårsak: vektetSjanse(faker, 0.5) ? 'Avsluttet' : null,
        historiskeUtbetalinger: fyllRandomListe<HistoriskUtbetaling>(() => getHistoriskUtbetaling(faker), 5),
        kommendeUtbetalinger: fyllRandomListe<KommendeUtbetaling>(() => getKommendeUtbetaling(faker), 5)
    };
}

function getArbeidsforholdMock(fødselsnummer: string): Arbeidsforhold {
    return {
        navn: faker.company.companyName(),
        kontonr: Number(faker.finance.account(11)).toString(),
        inntektsperiode: vektetSjanse(faker, 0.5) ? 'Månedlig' : 'Årlig',
        inntektForPerioden: Math.round(Number(faker.finance.amount(5000, 50000))),
        sykepengerFom: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        refusjonTom: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        refusjonstype: 'Ikke refusjon'
    };
}

function foreldrePengeType(): string {
    return navfaker.random.arrayElement([
        'Fødselspenger', 'Svangerskapspenger', 'Engangsstønad'
    ]);
}