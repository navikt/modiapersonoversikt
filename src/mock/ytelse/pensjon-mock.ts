import { fakerNB_NO as faker } from '@faker-js/faker';
import dayjs from 'dayjs';
import navfaker from 'nav-faker/dist/index';
import { statiskPensjonMock } from 'src/mock/ytelse/statiskPensjonMock';
import type { Pensjon, PensjonResource } from 'src/models/ytelse/pensjon';
import { backendDatoformat } from 'src/utils/date-utils';
import { aremark } from '../persondata/aremark';
import { fyllRandomListe } from '../utils/mock-utils';

export function getMockPensjon(fnr: string): PensjonResource {
    if (fnr === aremark.personIdent) {
        return [statiskPensjonMock];
    }

    faker.seed(Number(fnr));
    navfaker.seed(`${fnr}pensjon`);

    if (navfaker.random.vektetSjanse(0.3)) {
        return null;
    }

    return fyllRandomListe<Pensjon>(() => getMockPensjonYtelser(fnr), 3);
}

function getMockPensjonYtelser(fnr: string): Pensjon {
    faker.seed(Number(fnr));
    navfaker.seed(`${fnr}pensjon`);

    const fomDato = dayjs(faker.date.past({ years: 2 })).format(backendDatoformat);
    const tomDato = dayjs(fomDato).add(faker.number.int(40), 'days').format(backendDatoformat);

    return {
        sakid: faker.string.alphanumeric(8),
        fomDato,
        tomDato,
        enhetId: '0129',
        sakType: 'AFP',
        sakStatus: 'OPPRETTET'
    };
}
