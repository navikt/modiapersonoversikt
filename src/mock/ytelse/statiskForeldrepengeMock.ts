import { Fødsel } from '../../models/ytelse/foreldrepenger';

export const statiskKommendeUtbetaling = {
    vedtak: { fra: '2018-01-14', til: '2017-07-15' },
    utbetalingsgrad: 79,
    utbetalingsdato: '2018-06-25',
    bruttobeløp: 466,
    arbeidsgiverNavn: 'Vedvik - Lunde',
    arbeidsgiverOrgNr: '1234567890',
    arbeidsgiverKontonr: '549801574',
    dagsats: 705,
    saksbehandler: 'Mari Johannessen',
    type: 'KONTOØVERFØRING'
};

export const statiskForeldrepengeMock: Fødsel = {
    forelder: '10108000398',
    andreForeldersFnr: '03063850003',
    antallBarn: 5,
    barnetsFødselsdato: '2019-02-11',
    dekningsgrad: 34,
    fedrekvoteTom: '2019-02-10',
    mødrekvoteTom: null,
    foreldrepengetype: 'Fødselspenger',
    graderingsdager: 55,
    restDager: 41,
    rettighetFom: '2017-11-21',
    eldsteIdDato: null,
    foreldreAvSammeKjønn: null,
    periode: [
        {
            fødselsnummer: '10108000398',
            harAleneomsorgFar: null,
            harAleneomsorgMor: null,
            arbeidsprosentMor: 58,
            avslagsårsak: 'Avslag',
            avslått: 'Avslått',
            disponibelGradering: 9,
            erFedrekvote: null,
            erMødrekvote: null,
            forskyvelsesårsak1: null,
            forskyvelsesperiode1: { fra: '2014-10-08T14:22:55.629Z', til: '2014-08-22T10:50:40.296Z' },
            forskyvelsesårsak2: null,
            forskyvelsesperiode2: { fra: '2014-12-09T16:29:21.344Z', til: '2014-08-26T10:23:03.605Z' },
            foreldrepengerFom: '2017-04-29',
            midlertidigStansDato: '2019-02-10',
            morSituasjon: 'nobis sit quo et quis',
            rettTilFedrekvote: 'Rett til fedrekvote',
            rettTilMødrekvote: 'Ingen rett til mødrekvote',
            stansårsak: 'Avsluttet',
            kommendeUtbetalinger: [statiskKommendeUtbetaling]
        }
    ],
    slutt: null,
    arbeidsforhold: [
        {
            arbeidsgiverNavn: 'Bakke - Solheim',
            arbeidsgiverKontonr: '35179178779',
            inntektsperiode: 'Månedlig',
            inntektForPerioden: 40062,
            sykepengerFom: '2019-02-10',
            refusjonTom: '2019-02-11',
            refusjonstype: 'Ikke refusjon'
        },
        {
            arbeidsgiverNavn: 'Bakke - Solheim',
            arbeidsgiverKontonr: '35179178779',
            inntektsperiode: 'Månedlig',
            inntektForPerioden: 40062,
            sykepengerFom: '2019-02-10',
            refusjonTom: '2019-02-11',
            refusjonstype: 'Ikke refusjon'
        }
    ],
    erArbeidsgiverperiode: null,
    arbeidskategori: 'Arbeidstaker',
    termin: '2019-02-11'
};
