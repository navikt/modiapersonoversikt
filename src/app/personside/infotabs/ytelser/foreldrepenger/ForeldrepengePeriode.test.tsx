import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { statiskForeldrepengeMock } from '../../../../../mock/ytelse/statiskForeldrepengeMock';
import ForeldrepengePeriode from './ForeldrepengePeriode';
import TestProvider from '../../../../../test/Testprovider';

test('Foreldrepengeperiode matcher snapshot', () => {
    const testRettighet = statiskForeldrepengeMock;
    const result = renderer.create(
        <TestProvider>
            <ForeldrepengePeriode periode={testRettighet.periode[0]} periodenr={1} />
        </TestProvider>
    );

    expect(result).toMatchSnapshot();
});
