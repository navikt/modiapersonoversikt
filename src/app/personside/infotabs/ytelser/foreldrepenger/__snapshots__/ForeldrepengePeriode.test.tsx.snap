// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foreldrepengeperiode matcher snapshot 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c3 dt {
  color: #645f5a;
}

.c3 dd {
  font-weight: bold;
  margin-top: 0.3rem;
}

.c3 > div {
  margin: 0.9375rem 0;
  padding-right: 1rem;
  width: 13rem;
  overflow-wrap: break-word;
}

.c0 > *:first-child {
  background-color: #cce1f3;
  padding: 0.5rem 1.25rem;
}

.c0 ol {
  padding: 0;
  margin: 0;
}

.c5 {
  overflow: auto;
  box-shadow: 0 0 0 0.07rem rgba(0,0,0,0.25);
  border-radius: .25rem;
  background-color: white;
}

.c5 table {
  text-align: left;
  width: 100%;
}

.c5 table thead {
  border-bottom: 0.2rem solid rgba(0,0,0,0.15);
  font-weight: bold;
}

.c5 table th,
.c5 table td {
  padding: 0.7rem;
  vertical-align: middle;
}

.c5 table th {
  padding-bottom: 0.7rem;
}

.c5 table tr:nth-child(even) {
  background-color: rgba(0,0,0,0.1);
}

.c4 {
  border-radius: 0.5rem;
  background-color: #f1f0f0;
  box-shadow: inset 0 0 0 0.3rem white,inset 0 0 0 0.37rem rgba(0,0,0,0.15);
  padding: 1.25rem;
  white-space: nowrap;
}

.c4 > *:not(:last-child) {
  margin-bottom: 1rem;
}

.c2 {
  border-radius: 0.5rem;
  background-color: #f1f0f0;
  box-shadow: inset 0 0 0 0.3rem white,inset 0 0 0 0.37rem rgba(0,0,0,0.15);
  padding: 1.25rem;
}

.c6 {
  border-radius: 0.5rem;
  background-color: #f1f0f0;
  box-shadow: inset 0 0 0 0.3rem white,inset 0 0 0 0.37rem rgba(0,0,0,0.15);
  padding: 1.25rem;
}

.c6 > *:not(:last-child) {
  margin-bottom: 1rem;
}

.c1 {
  padding: .8rem;
}

<li
  className="c0"
>
  <h3
    className="typo-undertittel"
  >
    Periode 1 - 29.04.2017
  </h3>
  <article
    className="c1"
  >
    <div
      className="c2"
    >
      <h2
        className="typo-undertittel"
      >
        Om foreldrepengeperioden
      </h2>
      <dl
        className="c3"
      >
        <div>
          <dt
            className="typo-normal"
          >
            Arbeidsprosent mor
          </dt>
          <dd
            className="typo-normal"
          >
            58%
          </dd>
        </div>
        <div>
          <dt
            className="typo-normal"
          >
            Mors situasjon
          </dt>
          <dd
            className="typo-normal"
          >
            nobis sit quo et quis
          </dd>
        </div>
        <div>
          <dt
            className="typo-normal"
          >
            Disponibel gradering
          </dt>
          <dd
            className="typo-normal"
          >
            9%
          </dd>
        </div>
        <div>
          <dt
            className="typo-normal"
          >
            Forskyvelsesperiode
          </dt>
          <dd
            className="typo-normal"
          >
            08.10.2014 - 22.08.2014
          </dd>
        </div>
        <div>
          <dt
            className="typo-normal"
          >
            Andre forskyvelsesperiode
          </dt>
          <dd
            className="typo-normal"
          >
            09.12.2014 - 26.08.2014
          </dd>
        </div>
        <div>
          <dt
            className="typo-normal"
          >
            Midlertidig stans
          </dt>
          <dd
            className="typo-normal"
          >
            2019-02-10
          </dd>
        </div>
        <div>
          <dt
            className="typo-normal"
          >
            Stansårsak
          </dt>
          <dd
            className="typo-normal"
          >
            Avsluttet
          </dd>
        </div>
        <div>
          <dt
            className="typo-normal"
          >
            Avslått
          </dt>
          <dd
            className="typo-normal"
          >
            Avslått
          </dd>
        </div>
        <div>
          <dt
            className="typo-normal"
          >
            Mødrekvote
          </dt>
          <dd
            className="typo-normal"
          >
            —
          </dd>
        </div>
        <div>
          <dt
            className="typo-normal"
          >
            Rett til Mødrekvote
          </dt>
          <dd
            className="typo-normal"
          >
            Ingen rett til mødrekvote
          </dd>
        </div>
        <div>
          <dt
            className="typo-normal"
          >
            Aleneomsorg Mor
          </dt>
          <dd
            className="typo-normal"
          >
            —
          </dd>
        </div>
        <div>
          <dt
            className="typo-normal"
          >
            Fedrekvote
          </dt>
          <dd
            className="typo-normal"
          >
            —
          </dd>
        </div>
        <div>
          <dt
            className="typo-normal"
          >
            Rett til Fedrekvote
          </dt>
          <dd
            className="typo-normal"
          >
            Rett til fedrekvote
          </dd>
        </div>
        <div>
          <dt
            className="typo-normal"
          >
            Aleneomsorg Far
          </dt>
          <dd
            className="typo-normal"
          >
            —
          </dd>
        </div>
      </dl>
    </div>
    <section
      className="c4"
    >
      <h4
        className="typo-undertittel"
      >
        Kommende utbetalinger
      </h4>
      <div
        className="typo-normal c5"
      >
        <table
          role="table"
        >
          <thead
            role="rowgroup"
          >
            <tr
              role="row"
            >
              <th
                role="columnheader"
              >
                Utb.dato
              </th>
              <th
                role="columnheader"
              >
                Type
              </th>
              <th
                role="columnheader"
              >
                Periode
              </th>
              <th
                role="columnheader"
              >
                Utb.grad
              </th>
              <th
                role="columnheader"
              >
                Dagsats
              </th>
              <th
                role="columnheader"
              >
                Bruttobeløp
              </th>
              <th
                role="columnheader"
              >
                Arb.giver
              </th>
              <th
                role="columnheader"
              >
                Org.nummer
              </th>
            </tr>
          </thead>
          <tbody
            role="rowgroup"
          >
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                25.06.2018
              </td>
              <td
                role="cell"
              >
                KONTOØVERFØRING
              </td>
              <td
                role="cell"
              >
                14.01.2018 - 15.07.2017
              </td>
              <td
                role="cell"
              >
                79%
              </td>
              <td
                role="cell"
              >
                705.00 NOK
              </td>
              <td
                role="cell"
              >
                466.00 NOK
              </td>
              <td
                role="cell"
              >
                Vedvik - Lunde
              </td>
              <td
                role="cell"
              >
                1234567890
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </section>
    <section
      className="c6"
    >
      <h4
        className="typo-undertittel"
      >
        Utførte utbetalinger
      </h4>
    </section>
  </article>
</li>
`;
